pandas/core/strings.py

```diff
@@ -444,60 +442,33 @@ def str_match(arr, pat, case=True, flags=0, na=np.nan, as_indexer=False):
     flags : int, default 0 (no flags)
         re module flags, e.g. re.IGNORECASE
     na : default NaN, fill value for missing values.
-    as_indexer : False, by default, gives deprecated behavior better achieved
-        using str_extract. True return boolean indexer.
+    as_indexer : ignored
```

Comment:
Yes, I was not sure what to do with the keyword:

- remove -> but this will give a lot of errors, as up to now you had to specify the keyword to get the right behaviour (if you had groups in the regex)
- raise warning -> changed default to None, and when user specifies it, raise a warning to say it is ignored
- just ignore -> but then there is potential confusion on why it does nothing + people will keep specifying it although not needed

For now I choose the 'raise warning' option. The 'just ignore' would cause the less impact, but is also less informative.